src/time_calculations.rs

```diff
@@ -79,4 +80,39 @@ pub fn get_current_day_of_year() -> u32 {
 
 pub fn get_current_year() -> i32 {
     Utc::now().year()
-}
\ No newline at end of file
+}
+
+pub fn get_moon_data(dt: Date<Utc>) -> String {
+    let url = format!(
+        "https://api.usno.navy.mil/moon/phase?date={}&nump=1",
+        dt.format("%m/%d/%Y")
+        );
+    get(&url).unwrap().text().unwrap()
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    // this is a really slow test and will break without internet
+    fn test_retrieve_moon_data() {
+        // march 2019 supermoon
+        assert_eq!(get_moon_data(Utc.ymd(2019, 03, 17)), "{
```

Comment:
Would this test actually make an API call to `https://api.usno.navy.mil`? If so we should try to mock it if possible.

Reply:
It does and that is definitely something to change.  Yes, a mock is a good idea.  Another way around it is to break this function into two separate functions, one which simply fetches the data, and one which actually processes it.  That way we don't have to test the fetcher (because it's just a relay), and can quickly create a request to pass to the processor for testing.  It will speed up the tests and make them more-reliable and not dependent on external resources (like the network).  I will definitely refactor this test before the moon phase feature can land.

Reply:
That's awesome @robbystk! That would be great if you could do it. Thanks for your help!

Reply:
I discovered that mocking in rust is not as straightforward as in other languages, so I went the route of splitting up functionality.  There's now a function that gets the json from USNO with reqwest, and one that parses it with serde_json.  There are unit tests for the parsing function that are nice and fast.  I left an integrated test for the full process that actually hits the USNO's API, but it's ignored.

Reply:
Resolved with 7e923968fc42ac8c0023df0540784c22696cd050